sdk/python/kfp/v2/components/experimental/component_spec_test.py

```diff
@@ -14,109 +14,115 @@
+  #         ],
+  #         output_specs=[
+  #             component_spec.OutputSpec(name='output1', type="str"),
+  #         ],
+  #     )
 
-    with self.assertRaisesRegex(ValueError,
-                                'Non-unique output name "dupe_param"'):
-      component_spec.ComponentSpec(
-          name='component_1',
-          implementation=component_spec.ContainerSpec(
-              image='alpine',
-              commands=[
-                  'sh',
-                  '-c',
-                  'set -ex\necho "$0" > "$1"',
-                  component_spec.InputValuePlaceholder('input1'),
-                  component_spec.OutputPathPlaceholder('dupe_param'),
-              ],
-          ),
-          input_specs=[
-              component_spec.InputSpec(name='input1', type=str),
-          ],
-          output_specs=[
-              component_spec.OutputSpec(name='dupe_param', type=str),
-              component_spec.OutputSpec(name='dupe_param', type=str),
-          ],
-      )
+  #   with self.assertRaisesRegex(
+  #       ValueError,
+  #       'Argument "OutputPathPlaceholder\(name=\'output000\'\)" '
+  #       'references non-existing output.'):
+  #     component_spec.ComponentSpec(
+  #         name='component_1',
+  #         implementation=component_spec.ContainerSpec(
+  #             image='alpine',
+  #             commands=[
+  #                 'sh',
+  #                 '-c',
+  #                 'set -ex\necho "$0" > "$1"',
+  #                 component_spec.InputValuePlaceholder('input1'),
+  #                 component_spec.OutputPathPlaceholder('output000'),
+  #             ],
+  #         ),
+  #         input_specs=[
+  #             component_spec.InputSpec(name='input1', type=str),
+  #         ],
+  #         output_specs=[
+  #             component_spec.OutputSpec(name='output1', type=str),
+  #         ],
+  #     )
 
-  def test_component_spec_with_placeholder_referencing_nonexisting_input_output(
-      self):
-    with self.assertRaisesRegex(
-        ValueError,
-        'Argument "InputValuePlaceholder\(name=\'input000\'\)" '
-        'references non-existing input.'):
-      component_spec.ComponentSpec(
-          name='component_1',
-          implementation=component_spec.ContainerSpec(
-              image='alpine',
-              commands=[
-                  'sh',
-                  '-c',
-                  'set -ex\necho "$0" > "$1"',
-                  component_spec.InputValuePlaceholder('input000'),
-                  component_spec.OutputPathPlaceholder('output1'),
-              ],
-          ),
-          input_specs=[
-              component_spec.InputSpec(name='input1', type=str),
-          ],
-          output_specs=[
-              component_spec.OutputSpec(name='output1', type=str),
-          ],
-      )
+  def test_component_spec_save_to_component_yaml(self):
+    open_mock = mock_open()
+    expected_yaml = """annotations: null
+description: null
```

Comment:
Is that possible to not show null values here?  like when we save proto message to json, null values are not included by default.